pkg/backporter/httpcache_test.go

```diff
@@ -0,0 +1,109 @@
+package backporter
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"testing"
+)
+
+//bugzillaCache tests
+
+func TestBugzillaCache(t *testing.T) {
+	testcases := []struct {
+		name      string
+		cachedKey string
+		value     []byte
+		absentKey string
+	}{
+		{
+			name:      "key 1",
+			cachedKey: "key1",
+			value:     []byte("Test string"),
+			absentKey: "keyx",
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			bc := newBugzillaCache()
+			bc.set(tc.cachedKey, tc.value)
+			cachedVal, isCached := bc.get(tc.cachedKey)
+			if !isCached {
+				t.Errorf("key must be present in cache - %s", tc.cachedKey)
+			}
+			if !bytes.Equal(tc.value, cachedVal) {
+				t.Errorf("cached value wrong expected - %s, got - %s", tc.value, cachedVal)
+			}
+			bc.delete(tc.cachedKey)
+			_, isCachedAfterDelete := bc.get(tc.cachedKey)
+			if isCachedAfterDelete {
+				t.Errorf("key present in cache after deletion - %s", tc.cachedKey)
+			}
+			_, isAbsentKeyPresent := bc.get(tc.absentKey)
+			if isAbsentKeyPresent {
+				t.Errorf("invalid key marked ok - %s", tc.absentKey)
+			}
+
+		})
+	}
+}
+
+type transportMock struct {
+	response *http.Response
+	err      error
+}
+
+func (t transportMock) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	return t.response, t.err
+}
+
+func TestRoundTrip(t *testing.T) {
+	testcases := []struct {
+		name  string
+		tmock transportMock
+	}{
+		{
+			name: "Succesful request to mock server",
+			tmock: transportMock{
+				response: &http.Response{
+					Status:     http.StatusText(http.StatusOK),
+					StatusCode: http.StatusOK,
+					Body:       ioutil.NopCloser(bytes.NewBuffer([]byte("some data"))),
+				},
+				err: nil,
+			},
+		},
+		{
+			name: "Error request to mock server",
+			tmock: transportMock{
+				response: nil,
+				err:      errors.New("fake error"),
+			},
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			r, _ := http.NewRequest("GET", "http://somewhere.com/", nil)
+			tp := cachingTransport{
+				cache:     newBugzillaCache(),
+				transport: tc.tmock,
+			}
+			resp, err := tp.RoundTrip(r)
+			if err != tc.tmock.err {
+				t.Errorf("wrong error - expected %v, got %v", tc.tmock.err, err)
+			}
+			if resp != tc.tmock.response {
+				t.Errorf("incorrect response - expected %v, got %v", tc.tmock.response, resp)
+			}
+			resp, err = tp.RoundTrip(r)
```

Comment:
Why do another `RoundTrip` call here? we already get the `resp` on line 92?

Reply:
The first stage requests data which is uncached.
The next stage checks whether the data is cached and if the cached data is returned the second time that data is requested

Reply:
So two tests in one testcase?

Reply:
I can move that out if thats what is needed